picard/ui/metadatabox/edittagdialog.py

```diff
@@ -204,7 +204,7 @@ def _initialize_ui(self):
     def _setup_tag_combobox(self):
         """Set up the tag name combobox with supported tags."""
         self.default_tags = self._get_supported_tags()
-        visible_tags = (tn for tn in self.default_tags if not tn.startswith("~"))
+        visible_tags = (tn for tn in self.default_tags if not tn.startswith("~"))  # TODO: use TagVar.is_hidden?
```

Comment:
Yey, this could be futher refactored to actually use `TagVar` more directly instead of `tag_names`. But this could be done in a separate PR.